core/services/chainlink/config_capabilities.go

```diff
@@ -82,3 +87,50 @@ func (c *capabilitiesExternalRegistry) ChainID() string {
 func (c *capabilitiesExternalRegistry) Address() string {
 	return *c.c.Address
 }
+
+type gatewayConnectorConfig struct {
+	c toml.GatewayConnectorConfig
+}
+
+func (c *gatewayConnectorConfig) ChainIDForNodeKey() string {
+	return *c.c.ChainIDForNodeKey
+}
+func (c *gatewayConnectorConfig) NodeAddress() string {
+	return *c.c.NodeAddress
+}
+
+func (c *gatewayConnectorConfig) DonID() string {
+	return *c.c.DonID
+}
+
+func (c *gatewayConnectorConfig) Gateways() []config.ConnectorGatewayConfig {
+	t := []config.ConnectorGatewayConfig{}
```

Comment:
@DavidOrchard Does this work? I think it should panic in practice because you're initializing a zero-length array and then adding elements to indexes that don't exist.

You can get around this by constructing an array of the right length:
```
t := make([]config.ConnectorGatewayConfig, len(c.c.Gateways))

Reply:
I honestly don't know if it works or not, I do know the tests seem to pass.  But I don't know if there's a test that copies the structure.

Reply:
Please update the `config-full.toml` test files with example data for all fields so that we have coverage